src/applications/vaos/components/calendar/CalendarWidget.unit.spec.js

```diff
@@ -0,0 +1,199 @@
+
+    // Assert
+    await waitFor(() => {
+      expect(
+        screen.queryByText(
+          /You already have an appointment scheduled at this time. Please select another day or time/i,
+        ),
+      ).to.be.ok;
+    });
+
+    userEvent.click(
+      screen.getByText(
+        moment(now)
+          .hour(13)
+          .format('h:mm'),
+      ),
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.queryByText(
+          /You already have an appointment scheduled at this time. Please select another day or time/i,
+        ),
+      ).not.to.exist;
+    });
+  });
+
+  it('should not display scheduling duplicate appointment error message for cancelled appointment', async () => {
+    // Arrange
+    const now = moment()
+      .tz('America/Denver')
+      .add(1, 'month')
+      .startOf('month')
+      .hour(12);
+    const availableSlots = [
+      {
+        start: now.format('YYYY-MM-DDTHH:mm:ss'),
+      },
+      {
+        start: moment(now)
+          .hour(13)
+          .format('YYYY-MM-DDTHH:mm:ss'),
+      },
+    ];
+    const startMonth = now.format('YYYY-MM');
+    const submitted = false;
+    // Grouped by YYYY-MM
+    const upcomingAppointments = {
+      [now.format('YYYY-MM')]: [
+        {
+          start: now.format('YYYY-MM-DDTHH:mm:ss'),
+          status: 'cancelled',
+        },
+      ],
+    };
+
+    const TestPageStub = () => {
+      const dispatch = useDispatch();
+      const data = useSelector(state => state.newAppointment.data);
+      const isAppointmentSelectionError = useSelector(
+        state => state.newAppointment.isAppointmentSelectionError,
+      );
+
+      return (
+        <CalendarWidget
+          maxSelections={1}
+          availableSlots={availableSlots}
+          value={data?.selectedDates}
+          id="dateTime"
+          timezone="America/Denver"
+          additionalOptions={{
+            required: true,
+          }}
+          disabled={false}
+          onChange={(...args) => {
+            return dispatch(onCalendarChange(...args));
+          }}
+          minDate={moment(now).format('YYYY-MM-DD')}
+          maxDate={moment(now)
+            .add(395, 'days')
+            .format('YYYY-MM-DD')}
+          renderIndicator={_ => undefined}
+          required
+          requiredMessage="Please choose your preferred date and time for your appointment"
+          startMonth={startMonth}
+          showValidation={submitted && !data?.selectedDates?.length}
+          showWeekends
+          upcomingAppointments={upcomingAppointments}
+          isAppointmentSelectionError={isAppointmentSelectionError}
+        />
+      );
+    };
+
+    // Act
+    const store = createTestStore({});
+    const screen = renderWithStoreAndRouter(<TestPageStub />, {
+      store,
+    });
+
+    userEvent.click(screen.getByText('1'));
+    userEvent.click(screen.getByText(now.format('h:mm')));
+
+    // Assert
+    await waitFor(() => {
```

Comment:
Is this where you mentioned needing to add the await waitFor to get the tests to pass in the pipeline? I don't know if it is necessary anymore or if it was a weird side effect. I'd be curious if the pipeline would pass without them. I removed them locally and the isolated test passed.

Reply:
Yes. I tested the changes in CI before the tests where skipped and they passed. Basically, the expect statement in the failing test are testing async code. These types of errors always pass locally since are machines are faster than the CI build.